wpilibc/src/main/native/include/frc/smartdashboard/SendableChooser.inc

```diff
@@ -95,12 +101,20 @@ void SendableChooser<T>::InitSendable(nt::NTSendableBuilder& builder) {
       nullptr);
   builder.AddStringProperty(kSelected, nullptr,
                             [=, this](std::string_view val) {
-                              std::scoped_lock lock(m_mutex);
-                              m_haveSelected = true;
-                              m_selected = val;
-                              for (auto& pub : m_activePubs) {
-                                pub.Set(val);
+                              {
+                                std::scoped_lock lock(m_mutex);
+                                m_haveSelected = true;
+                                m_selected = val;
+                                for (auto& pub : m_activePubs) {
+                                  pub.Set(val);
+                                }
+                              }
+                              if (m_previousVal != val && m_listener) {
+                                auto choice = m_choices[val];
+                                auto listener = m_listener;
+                                listener(choice);
                               }
+                              m_previousVal = val;
```

Comment:
This is unsafe, you're not holding the mutex when accessing several m_ variables.  Needs to be this:
```
                              T choice;
                              std::function<void(T)> listener;
                              {
                                std::scoped_lock lock(m_mutex);
                                m_haveSelected = true;
                                m_selected = val;
                                for (auto& pub : m_activePubs) {
                                  pub.Set(val);
                                }
                                if (m_previousVal != val && m_listener) {
                                  choice = m_choices[val];
                                  listener = m_listener;
                                }
                                m_previousVal = val;
                              }
                              if (listener) {
                                listener(choice);
                              }
```